app/code/Magento/CatalogSearch/Model/Layer/Search/Plugin/CollectionFilter.php

```diff
@@ -15,34 +15,34 @@ class CollectionFilter
      * @var \Magento\Search\Model\QueryFactory
      */
     protected $queryFactory;
-
     /**
      * @param QueryFactory $queryFactory
      */
     public function __construct(QueryFactory $queryFactory)
     {
         $this->queryFactory = $queryFactory;
     }
-
     /**
      * Add search filter criteria to search collection
      *
      * @param \Magento\Catalog\Model\Layer\Search\CollectionFilter $subject
-     * @param null $result
+     * @param \Closure $proceed
      * @param \Magento\CatalogSearch\Model\ResourceModel\Fulltext\Collection $collection
      * @param Category $category
      * @return void
      * @SuppressWarnings(PHPMD.UnusedFormalParameter)
      */
-    public function afterFilter(
+    public function aroundFilter(
         \Magento\Catalog\Model\Layer\Search\CollectionFilter $subject,
-        $result,
+        \Closure $proceed,
         $collection,
         Category $category
     ) {
         /** @var \Magento\Search\Model\Query $query */
         $query = $this->queryFactory->get();
+
         if (!$query->isQueryTextShort()) {
+            $proceed($collection, $category);
```

Comment:
If `$query->isQueryTextShort()` is false you are not calling the `$proceed`. I know you did to avoid having the collection filtered, but not calling it may prevent from calling other plugins on the same method.

IMHO you could try keeping the after plugin and adding a `before` plugin passing an empty collection the collection if the lenght criteria is not satisfied.

This could be a rough idea of what I mean:
```
public function beforeFilter(
    \Magento\Catalog\Model\Layer\Search\CollectionFilter $subject,
    $collection,
    \Magento\Catalog\Model\Category $category
) {
    /** @var \Magento\Search\Model\Query $query */
    $query = $this->queryFactory->get();
    if ($query->isQueryTextShort()) {
        return [$this->collectionFactory->create(), $category];
    }

    return [$collection, $category];
}
```

It is not a clean solution as the one you provided, but it will not break other plugins.
My implementation is just an example, if you have a better idea it is appreciated.